src/main/java/uk/nhs/prm/e2etests/service/PdsAdaptorService.java

```diff
@@ -1,40 +1,62 @@
 package uk.nhs.prm.e2etests.service;
 
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.http.*;
+import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
+import uk.nhs.prm.e2etests.exception.InvalidPdsAdaptorUsernameException;
 import uk.nhs.prm.e2etests.model.request.PdsAdaptorRequest;
 import uk.nhs.prm.e2etests.model.response.PdsAdaptorResponse;
+import uk.nhs.prm.e2etests.property.PdsAdaptorProperties;
 
-// TODO PRMT-3523 -> THIS SHOULD BE A SERVICE
+@Log4j2
+@ConditionalOnProperty(
+        prefix = "test.pds",
+        name = { "username" }
+)
+@Service
 public class PdsAdaptorService {
 
     private final String apiKey;
     private final String patientRootUrl;
     private final String username;
+
     private final RestTemplate restTemplate = new RestTemplate();
 
-    public PdsAdaptorService(String username, String apiKey, String pdsAdaptorUrl) {
+    @Autowired
+    public PdsAdaptorService(PdsAdaptorProperties pdsAdaptorProperties, @Value("${test.pds.username}") String username) {
+        this.patientRootUrl = pdsAdaptorProperties.getPdsAdaptorUrl();
         this.username = username;
-        this.apiKey = apiKey;
-        this.patientRootUrl = pdsAdaptorUrl;
+
+        switch (username) {
+            case "live-test" -> this.apiKey = pdsAdaptorProperties.getLiveTestApiKey();
+            case "e2e-test" -> this.apiKey = pdsAdaptorProperties.getE2eTestApiKey();
+            case "performance-test" -> this.apiKey = pdsAdaptorProperties.getPerformanceApiKey();
```

Comment:
Could we possibly replace the 3 strings here with Enum class?  (Although I'm not sure how)